src/assets/wise5/directives/student-summary-display/student-summary-display.component.ts

```diff
@@ -51,4 +75,200 @@ export class StudentSummaryDisplay extends SummaryDisplay {
+    } else {
+      return this.getLatestStudentWork();
+    }
+  }
+
+  private getDummyStudentWorkForVLEPreview(
+    nodeId: string,
+    componentId: string
+  ): Observable<ComponentState[]> {
+    const componentStates = this.createDummyComponentStates();
+    const componentState = (
+      this.dataService as StudentDataService
+    ).getLatestComponentStateByNodeIdAndComponentId(nodeId, componentId);
+    if (componentState != null) {
+      componentStates.push(componentState);
+    }
+    return of(componentStates);
+  }
+
+  private getDummyStudentScoresForVLEPreview(): Observable<Annotation[]> {
+    const annotations = this.createDummyScoreAnnotations();
+    const annotation = this.getLatestScoreAnnotationForWorkgroup();
+    if (annotation != null) {
+      annotations.push(annotation);
+    }
+    return of(annotations);
+  }
+
+  private getDummyStudentWorkForAuthoringPreview(): Observable<ComponentState[]> {
+    return of(this.createDummyComponentStates());
+  }
+
+  private getDummyStudentScoresForAuthoringPreview(): Observable<Annotation[]> {
+    return of(this.createDummyScoreAnnotations());
+  }
+
+  private createDummyComponentStates(): DummyComponentState[] {
+    const dummyComponentStates = [];
+    for (let dummyCounter = 0; dummyCounter < this.numDummySamples; dummyCounter++) {
+      dummyComponentStates.push(this.createDummyComponentState());
+    }
+    return dummyComponentStates;
+  }
+
+  private createDummyComponentState(): DummyComponentState {
+    if (this.otherComponentType === 'MultipleChoice') {
+      return this.createDummyMultipleChoiceComponentState(
+        this.otherComponent as MultipleChoiceContent
+      );
+    } else if (this.otherComponentType === 'Table') {
+      return this.createDummyTableComponentState();
+    }
+  }
+
+  private createDummyMultipleChoiceComponentState(componentState: MultipleChoiceContent): any {
+    const choices = componentState.choices;
+    return {
+      studentData: {
+        studentChoices: [{ id: this.getRandomChoice(choices).id }]
+      }
+    };
+  }
+
+  private createDummyTableComponentState(): any {
+    if (this.isAuthoringPreview()) {
+      return {
+        studentData: {
+          tableData: this.getDummyTableData()
+        }
+      };
+    } else {
+      return {
+        studentData: {
+          tableData: this.getDummyTableDataSimilarToLatestComponentState()
+        }
+      };
+    }
+  }
+
+  private getDummyTableData(): any[] {
+    return [
+      [{ text: 'Trait' }, { text: 'Count' }],
+      [{ text: 'Blue' }, { text: '3' }],
+      [{ text: 'Green' }, { text: '2' }],
+      [{ text: 'Red' }, { text: '1' }]
+    ];
+  }
+
+  private getDummyTableDataSimilarToLatestComponentState(): any[] {
+    let tableData = [];
+    const componentState = (
+      this.dataService as StudentDataService
+    ).getLatestComponentStateByNodeIdAndComponentId(this.nodeId, this.componentId);
+    if (componentState != null) {
```

Comment:
Safe to remove null check? Since this function will only be called after the student submits to the other component, will this componentState ever be null? 
```suggestion
```